app/controllers/trainees/confirm_deferrals_controller.rb

```diff
@@ -3,6 +3,7 @@
 module Trainees
   class ConfirmDeferralsController < ApplicationController
     before_action :authorize_trainee
+    before_action :ensure_trainee_is_not_draft!
```

Comment:
There's a card in the code review backlog (cc @gpeng  and @defong): https://trello.com/c/CEd6xa0D/1052-ensuretraineeisdraft-doesnt-belong-in-the-applicationcontroller

If feels like we should be using policies (Pundit) here instead, and in fact we already have a trainee policy for `defer?`. For an example of how it's used, see the deferrals controller - `authorize(trainee, :defer?)`.

You could update the `def authorize_trainee` method in this controller to use this. That way the trainee state check is more strict than `not draft` too - the policy states that the trainee must be `(trainee.submitted_for_trn? || trainee.trn_received?)`

Reply:
Updated the method to use `authorize(trainee, :defer?)`